src/components/Pages/Form/Form.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  Box,
+  Button,
+  Chip,
+  FormControl,
+  FormGroup,
+  FormHelperText,
+  IconButton,
+  Input,
+  InputLabel,
+  Typography,
+} from "@mui/material";
+import { useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { setNewHabit } from "../../../app/features/habit/habitSlice";
+import AddCircleIcon from "@mui/icons-material/AddCircle";
+import { createHabitData } from "../../../utilities/createHabitData";
+import { Habit } from "../../../app/features/habit/habitTypes";
+import lightFormat from "date-fns/lightFormat";
+
+const HabitForm = () => {
+  const [rewards, setRewards] = useState<{ label: string; id: number }[]>([]);
+  const [rewardInput, setRewardInput] = useState("");
+  const [goalInput, setGoalInput] = useState("");
+  const [schemaInput, setSchemaInput] = useState("");
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  return (
+    <Box sx={{ padding: "4rem", maxWidth: "50%" }}>
+      <form
+        onSubmit={(e) => {
+          const newHabit: Habit = {
+            goal: goalInput,
+            schema: schemaInput,
+            rewards,
+            id: Math.floor(Math.random() * 100000),
```

Comment:
In most cases it's better to use strings for IDs. Im the current implementation the number of possible combinations is (10^5) and that's quite low.

Reply:
I'd use something like `UUID v4`

Reply:
`crypto.randomUUID()`